src/Orchard/Tasks/Locking/Services/DistributedLockService.cs

```diff
@@ -13,128 +14,120 @@
 namespace Orchard.Tasks.Locking.Services {
 
     public class DistributedLockService : Component, IDistributedLockService {
-        private readonly IMachineNameProvider _machineNameProvider;
-        private readonly ILifetimeScope _lifetimeScope;
+        private readonly IRepository<DistributedLockRecord> _repository;
         private readonly IClock _clock;
-        private readonly IThreadProvider _threadProvider;
+        private readonly IMachineNameProvider _machineNameProvider;
+        private readonly ShellSettings _shellSettings;
 
-        public DistributedLockService(IMachineNameProvider machineNameProvider, IThreadProvider threadProvider, ILifetimeScope lifetimeScope, IClock clock) {
-            _machineNameProvider = machineNameProvider;
-            _lifetimeScope = lifetimeScope;
-            _clock = clock;
-            _threadProvider = threadProvider;
-        }
+        private readonly ConcurrentDictionary<string, DistributedLock> _locks = new ConcurrentDictionary<string, DistributedLock>();
 
-        public bool TryAcquireLockForMachine(string name, TimeSpan? maxValidFor, TimeSpan? timeout, out DistributedLock @lock) {
-            return TryAcquireLock(name, maxValidFor, timeout, GetMachineName(), null, out @lock);
-        }
+        public bool DisableMonitorLock { get; set; }
 
-        public DistributedLock AcquireLockForMachine(string name, TimeSpan? maxValidFor, TimeSpan? timeout) {
-            return AcquireLock(name, maxValidFor, timeout, GetMachineName(), null);
-        }
+        public DistributedLockService(
+            IMachineNameProvider machineNameProvider,
+            IRepository<DistributedLockRecord> repository,
+            IClock clock,
+            ShellSettings shellSettings) {
+            _clock = clock;
+            _repository = repository;
 
-        public bool TryAcquireLockForThread(string name, TimeSpan? maxValidFor, TimeSpan? timeout, out DistributedLock @lock) {
-            return TryAcquireLock(name, maxValidFor, timeout, GetMachineName(), GetThreadId(), out @lock);
+            _shellSettings = shellSettings;
+            _machineNameProvider = machineNameProvider;
         }
 
-        public DistributedLock AcquireLockForThread(string name, TimeSpan? maxValidFor, TimeSpan? timeout) {
-            return AcquireLock(name, maxValidFor, timeout, GetMachineName(), GetThreadId());
+        public bool TryAcquireLock(string name, TimeSpan? maxValidFor, TimeSpan? timeout, out IDistributedLock l) {
+            try {
+                l = AcquireLock(name, maxValidFor, timeout);
+                return l != null;
+            }
+            catch {
+                l = null;
+                return false;
+            }
         }
 
-        public void ReleaseLock(DistributedLock @lock) {
-            var childLifetimeScope = CreateChildLifetimeScope(@lock.Name);
```

Comment:
The reason we created a child scope is so that we could create a new transaction with a `ReadCommitted` isolation level (without affecting the isolation level of the current transaction).
Do you think that was unnecessary? What if two machines try and acquire a lock at the same time and the system is configured to use `ReadUncommitted`?